gdxsv/lbs_discord.go

```diff
@@ -0,0 +1,494 @@
+		}
+		logger.Info("Lobby string size reduced!")
+	}
+	embedAccumulatedBattleStringLength := len(accumulatedBattlePeersString)
+	logger.Info("Battle string Size", zap.Int("length", embedAccumulatedBattleStringLength))
+	if embedAccumulatedBattleStringLength > 6000 {
+		for _, b := range battle {
+			b.RenpoPeers = reducePeerStringSize(b.RenpoPeers)
+			b.ZeonPeers = reducePeerStringSize(b.ZeonPeers)
+		}
+		logger.Info("Battle string size reduced!")
+	}
+
+	//
+	// Start to create JSON payload
+	//
+	payload := new(statusPayload)
+	payload.BotName = "Live Status"
+
+	//
+	//1st Embed, online count
+	//
+	payload.Embed = append(payload.Embed, &discordEmbed{
+		Title:     fmt.Sprintf("**同時接続数 %d 人 **", lobbyPeerCount+battlePeerCount),
+		Color:     52224,
+		Footer:    &discordEmbedFooter{Text: "🕒"},
+		Timestamp: fmt.Sprintf(time.Now().UTC().Format("2006-01-02T15:04:05.000Z")),
+	})
+
+	//
+	//2nd Embed, lobby count
+	//
+
+	//1st Field is always Plaza
+	var lobbyFields []*discordEmbedField
+	if plazaPeerCount > 0 {
+		for i, v := range splitEmbedFieldValues(plazaPeers) {
+			name := "⠀"
+			if i == 0 {
+				name = fmt.Sprintf("**Plaza － %d 人**", plazaPeerCount)
+			}
+
+			lobbyFields = append(lobbyFields, &discordEmbedField{
+				Name:  name,
+				Value: v,
+			})
+		}
+	}
+
+	//Following Fields would be all lobbies
+	//Use sortedKeys to fix the ordering
+	for _, i := range sortedKeys(lobby) {
+		l := lobby[i]
+
+		value := addBlankIfRequired(l.RenpoPeers+l.ZeonPeers) + addBlankIfRequired(l.RenpoRoomPeers+l.ZeonRoomPeers) + addBlankIfRequired(l.NoForcePeers)
+
+		for i, v := range splitEmbedFieldValues(value) {
+			name := "⠀"
+			if i == 0 {
+				name = fmt.Sprintf("**%s － %d 人\n%s %s**", l.Name, l.Count, l.RegionName, l.Comment)
+			}
+
+			lobbyFields = append(lobbyFields, &discordEmbedField{
+				Name:  name,
+				Value: v,
+			})
+		}
+
+	}
+	for i, fields := range splitEmbedFields(lobbyFields) {
+		description := ""
+		if i == 0 {
+			description = fmt.Sprintf("🌐 **待機中 %d 人**", lobbyPeerCount)
+		}
+		payload.Embed = append(payload.Embed, &discordEmbed{
+			Description: description,
+			Color:       24041,
+			Fields:      fields,
+		})
+	}
+
+	//
+	//3rd Embed, battle count
+	//
+	var battleFields []*discordEmbedField
+	for _, b := range battle {
+
+		value := addBlankIfRequired(b.RenpoPeers + b.ZeonPeers)
+
+		for i, v := range splitEmbedFieldValues(value) {
+			name := "⠀"
+			if i == 0 {
+				name = b.RegionName
+			}
+
+			battleFields = append(battleFields, &discordEmbedField{
+				Name:  name,
+				Value: v,
+			})
+		}
+	}
+	for i, fields := range splitEmbedFields(battleFields) {
+		description := ""
+		if i == 0 {
+			description = fmt.Sprintf("💥 **戦闘中 %d 人**", battlePeerCount)
+		}
+		payload.Embed = append(payload.Embed, &discordEmbed{
+			Description: description,
+			Color:       13179394,
+			Fields:      fields,
+		})
+	}
+
+	//
+	// Create the json
+	//
+	var jsonData []byte
+	jsonData, err := json.Marshal(payload)
```

Comment:
This wastefully allocates memory and copies.
Let's use `bytes.Buffer` and `json.Encoder`.

Reply:
[Done](https://github.com/inada-s/gdxsv/pull/66/files#diff-bbc193ec510092c18d1667da1c5232f4dfe8137c20b8f99db6b3c1fe606449cdR423-R424)
```go
buffer := &bytes.Buffer{}
encoder := json.NewEncoder(buffer)
```